utilities/titandb/titan_db_test.cc

```diff
@@ -237,19 +298,56 @@ TEST_F(TitanDBTest, DbIter) {
 }
 
 TEST_F(TitanDBTest, Snapshot) {
-    Open();
-    std::map<std::string, std::string> data;
-    Put(1, &data);
-    ASSERT_EQ(1, data.size());
+  Open();
+  std::map<std::string, std::string> data;
+  Put(1, &data);
+  ASSERT_EQ(1, data.size());
 
-    const Snapshot* snapshot(db_->GetSnapshot());
-    ReadOptions ropts;
-    ropts.snapshot = snapshot;
+  const Snapshot* snapshot(db_->GetSnapshot());
+  ReadOptions ropts;
+  ropts.snapshot = snapshot;
 
-    VerifyDB(data, ropts);
-    Flush();
-    VerifyDB(data, ropts);
-    db_->ReleaseSnapshot(snapshot);
+  VerifyDB(data, ropts);
+  Flush();
+  VerifyDB(data, ropts);
+  db_->ReleaseSnapshot(snapshot);
+}
+
+TEST_F(TitanDBTest, IngestExternalFiles) {
+  Open();
+  SstFileWriter sst_file_writer(EnvOptions(), options_);
+  ASSERT_EQ(sst_file_writer.FileSize(), 0);
+
+  const uint64_t kNumEntries = 100;
+  std::map<std::string, std::string> data;
+  for (uint64_t i = 1; i <= kNumEntries; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries, data.size());
+  Flush();
+  VerifyDB(data);
+
+  const uint64_t kNumIngestedEntries = 100;
+  std::string sst_file = options_.dirname + "/for_ingest.sst";
+  ASSERT_OK(sst_file_writer.Open(sst_file));
+  for (uint64_t i = 1; i <= kNumIngestedEntries; i++) {
+    std::string key = GenKey(kNumEntries + i);
+    std::string value = GenValue(kNumEntries + i);
+    ASSERT_OK(sst_file_writer.Put(key, value));
+    data.emplace(key, value);
+  }
+  ASSERT_OK(sst_file_writer.Finish());
+  IngestExternalFileOptions ifo;
+  ASSERT_OK(db_->IngestExternalFile({sst_file}, ifo));
+  Flush();
+  VerifyDB(data);
+  VerifyBlob(data, false);
```

Comment:
it may be a bit more clear if we remove the `compacted` param from `VerifyBlob`, and simply verify number of blob files here: `ASSERT_EQ(0, GetBlobStorage()->NumBlobFiles())`